src/applications/auth/containers/AuthApp.jsx

```diff
@@ -25,9 +29,15 @@ export class AuthApp extends React.Component {
 
   handleAuthSuccess = payload => {
     setupProfileSession(payload);
-    const returnUrl = sessionStorage.getItem('authReturnUrl');
-    sessionStorage.removeItem('authReturnUrl');
-    window.location = returnUrl || '/';
+    this.redirect();
+  };
+
+  redirect = () => {
+    const returnUrl = sessionStorage.getItem(authnSettings.RETURN_URL) || '';
+    sessionStorage.removeItem(authnSettings.RETURN_URL);
+
+    window.location =
+      (!returnUrl.match(window.location.pathname) && returnUrl) || '/';
```

Comment:
This ensures that we don't redirect back to the callback page if logging in after an error.